game/scripts/vscripts/units/swiper.lua

```diff
@@ -0,0 +1,213 @@
+require('abilities/swiper/boss_swiper_swipe')
+
+function Spawn( entityKeyValues )
+	if thisEntity == nil then
+		return
+	end
+
+	if IsServer() == false then
+		return
+	end
+
+	thisEntity.hThrustAbility = thisEntity:FindAbilityByName( "boss_swiper_thrust" )
+	thisEntity.hFrontswipeAbility = thisEntity:FindAbilityByName( "boss_swiper_frontswipe" )
+	thisEntity.hBackswipeAbility = thisEntity:FindAbilityByName( "boss_swiper_backswipe" )
+	thisEntity.hReapersRushAbility = thisEntity:FindAbilityByName( "boss_swiper_reapers_rush" )
+
+	thisEntity:SetContextThink( "SwiperBossThink", SwiperBossThink, 1 )
+end
+
+function SwiperBossThink()
+	if GameRules:IsGamePaused() == true or GameRules:State_Get() == DOTA_GAMERULES_STATE_POST_GAME or thisEntity:IsAlive() == false then
+		return 1
+	end
+
+	-- if not thisEntity:IsIdle() then
+	-- 	return 1.0
+	-- end
+
+	if not thisEntity.bInitialized then
+		thisEntity.vInitialSpawnPos = thisEntity:GetOrigin()
+		thisEntity.bInitialized = true
+	end
+
+	local enemies = FindUnitsInRadius(
+		thisEntity:GetTeamNumber(),
+		thisEntity:GetOrigin(), nil,
+		1000,
+		DOTA_UNIT_TARGET_TEAM_ENEMY,
+		DOTA_UNIT_TARGET_HERO,
+		DOTA_UNIT_TARGET_FLAG_MAGIC_IMMUNE_ENEMIES + DOTA_UNIT_TARGET_FLAG_FOW_VISIBLE,
+		FIND_CLOSEST,
+		false
+	)
+
+	if #enemies == 0 then
+		return RetreatHome()
+	end
+
+	local canUseRush = true
+	local canUseTeleport = false
+
+	if thisEntity:GetHealth() / thisEntity:GetMaxHealth() > 0.75 then -- phase 1
+		canUseRush = false
+	elseif thisEntity:GetHealth() / thisEntity:GetMaxHealth() > 0.5 then -- phase 2
+		canUseRush = true
+	else
+		canUseTeleport = true
+	end
+
+	-- canUseRush = true
+
+	-- Swipe
+
+	local swipeRange = thisEntity.hFrontswipeAbility:GetCastRange(thisEntity:GetAbsOrigin(), thisEntity)
```

Comment:
this is nitpicking but if we decide to change an ability in the future would be nince to the AI not to break, null check the abilities if you can